dev-requirements.txt

```diff
@@ -1,7 +1,7 @@
 # install latest changes in dbt-core
 # TODO: how to automate switching from develop to version branches?
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+git+https://github.com/dbt-labs/dbt-core.git@jerco/mv-onschemachange-adapter-zone#egg=dbt-core&subdirectory=core
+git+https://github.com/dbt-labs/dbt-core.git@jerco/mv-onschemachange-adapter-zone#egg=dbt-tests-adapter&subdirectory=tests/adapter
```

Comment:
[TODO before merging] Change back to `main` after https://github.com/dbt-labs/dbt-core/pull/6618 has been merged